tests/code/test_transcripts.py

```diff
@@ -1,63 +1,197 @@
-    assert transcript[3] == ['Empty Bowls', 'Spring A 2021', 10.0]
-
-    user = "khatts"
-    transcript = getProgramTranscript(user)
-    assert transcript[0] == ['Adopt A Grandparent', 'Summer 2021', 3.0] #Program that has events from different term.
-    assert transcript[2] == ['Empty Bowls', 'Spring A 2021', 5.0]
-
-    user = "ramsayb2" #user who's not involved in any program
-    transcript = getProgramTranscript(user)
-    assert transcript == []
+def testingProgram():
+
+
+    username = "namet"
+    adminName = "ramsayb2"
+
+    newEvent = Event.create(name = "Test Program Event",
+                              term = 1,
+                              description= "Event for testing",
+                              timeStart = "18:00:00",
+                              timeEnd = "21:00:00",
+                              location = "The testing lab",
+                              isRecurring = 0,
+                              isRsvpRequired = 0,
+                              isPrerequisiteForProgram = 0,
+                              isTraining = 0,
+                              isService = 0,
+                              startDate =  2021-12-12,
+                              endDate =  2021-12-13)
+
+    programEvent = ProgramEvent.create(program=1, event=newEvent)
+
+    testingProgramEvent = Event.get(Event.name == "Test program Event")
+
+    addVolunteerToEvent('namet', testingProgramEvent.id, 2)
+
+    testingProgramExist = getProgramTranscript(username)
+    testingProgramNotExist = getProgramTranscript(adminName)
+
+    assert testingProgramNotExist.exists() == False
+    assert [program.program.programName == "Test Program Event" for program in testingProgramExist]
+    assert testingProgramExist.exists()
+
+
+
+@pytest.mark.integration
+def testingTotalHours():
+
+    totalHours = getTotalHours("namet")
+
+    assert totalHours == 9
+
+    # delete training
+    testingTrainingEvent = Event.get(Event.name == "Test Training Event")
+    deleteEvent(testingTrainingEvent)
+    assert Event.get_or_none(Event.id == testingTrainingEvent) is None
+    # delete bonner
+    testingBonnerEvent = Event.get(Event.name == "Test Bonner Event")
+    deleteEvent(testingBonnerEvent)
+    assert Event.get_or_none(Event.id == testingBonnerEvent) is None
+    # delete courses
+    testingCourse = Course.get(Course.courseName == "Test Course")
+    testingCourse.delete_instance(recursive = True, delete_nullable = True)
+    assert Course.get_or_none(Course.id == testingCourse.id) is None
+    # delete program
+    testingProgramEvent = Event.get(Event.name == "Test program Event")
+    deleteEvent(testingProgramEvent)
+    assert Event.get_or_none(Event.id == testingProgramEvent) is None
+    # delete user
+    user = User.get(User.username == "namet")
+    user.delete_instance(recursive = True, delete_nullable = True)
+    assert User.get_or_none(User.username == "namet") is None
```

Comment:
Managing your test data this way has created dependencies between your tests. If earlier tests fail or are run independently they will leave bad data in the database. If the tests run in a different order they could fail (this one will, in any case).

If you want to have the same data set for a group of tests, but don't want to use the test data in the database, use `setup` and `teardown` methods (search for how to do that with pytest).